packages/ui/src/RangeBar.tsx

```diff
@@ -0,0 +1,132 @@
+import React, {
+  useContext,
+  useState,
+  useEffect,
+  ForwardedRef,
+  forwardRef,
+  HTMLProps,
+} from 'react'
+import styled from 'styled-components'
+
+import { Text } from './Text'
+
+export interface RangeBarProps extends HTMLProps<HTMLInputElement> {
+  /**
+   * Step value of scroll bar
+   */
+  stepvalue?: number
+}
+
+/**
+ * Height of the scroll bar wrapper container
+ */
+const containerHeight = 70
+
+export const RangeBar = forwardRef(
+  (
+    {
+      type = 'range',
+      className = '',
+      disabled = false,
+      stepvalue,
+      min,
+      max,
+      ...props
+    }: RangeBarProps,
+    ref: ForwardedRef<HTMLInputElement>,
+  ) => {
+    const [valuePercent, setValuePercent] = useState(0)
```

Comment:
we should avoid having a controlled-state components, and since this is a simple component, no need to have a controlled-state.

```tsx
const max = Number(props.max)
const min = Number(props.min)
const percentage = numberToPercentage(props.value / max)

```

everything else here we can remove like `handleRangeChange,calculatePercent,setValuePercent,rangeValue` and `useEffect`s